mongomock/collection.py

```diff
@@ -1045,7 +1046,7 @@ def _update_document_single_field(self, doc, field_name, field_value, updater):
         field_name = field_name_parts[-1]
         if isinstance(doc, list):
             try:
-                doc[int(field_name)] = field_value
+                updater(doc, field_name, field_value)
```

Comment:
I'd rather not have this line copied from below. I suggest you handle the case where one silence the`IndexError` in the updater functions.

Edit: after reading your code below, it seems it's already handled and that you won't have any IndexError raised, so no need to differentiate the cases.

Reply:
ok